packages/js-client/src/types.ts

```diff
@@ -0,0 +1,22 @@
+import type { operations } from '@netlify/open-api'
+
+type OperationParams<K extends keyof operations> = 'parameters' extends keyof operations[K]
+  ? operations[K]['parameters']
+  : undefined
+type SuccessHttpStatusCodes = 200 | 201 | 202 | 203 | 204 | 205 | 206 | 207 | 208 | 226
+
+type OperationResponse<K extends keyof operations> = 'responses' extends keyof operations[K]
```

Comment:
todo: Try this out and then handle other response types like errors since fetch wont throw errors for non 2xx status codes.